File: backend/app/schemas/watchlist.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field


class WatchlistCreate(BaseModel):
    target_type: Literal["industry", "sector", "stock"]
    target_code: str = Field(min_length=1, max_length=50)
    target_name: str | None = None
    note: str | None = None


class WatchlistResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    target_type: str
    target_code: str
    target_name: str | None = None
    note: str | None = None
    created_at: datetime | None = None


class EventBrief(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    event_type: str
    title: str
    content: str | None = None
    related_type: str | None = None
    related_code: str | None = None
    report_id: int | None = None
    severity: str
    occurred_at: datetime | None = None


class NotificationResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    event_id: int
    is_read: bool
    created_at: datetime | None = None
    event: EventBrief | None = None


class NotificationListResponse(BaseModel):
    items: list[NotificationResponse]
    total: int
    page: int
    page_size: int


class UnreadCountResponse(BaseModel):
    count: int

File: frontend/src/types/watchlist.ts
export interface WatchlistItem {
  id: number;
  target_type: "industry" | "sector" | "stock";
  target_code: string;
  target_name?: string | null;
  note?: string | null;
  created_at?: string | null;
}

export interface EventBrief {
  id: number;
  event_type: string;
  title: string;
  content?: string | null;
  related_type?: string | null;
  related_code?: string | null;
  report_id?: number | null;
  severity: string;
  occurred_at?: string | null;
}

export interface NotificationItem {
  id: number;
  event_id: number;
  is_read: boolean;
  created_at?: string | null;
  event?: EventBrief | null;
}

export interface NotificationListResponse {
  items: NotificationItem[];
  total: number;
  page: number;
  page_size: number;
}
